klient/k8s/resource/resource.go

```diff
@@ -41,11 +41,11 @@ type Resources struct {
 	client cr.Client
 }
 
-// Res instantiates the controller runtime client
+// NewResource instantiates the controller runtime client
 // object. User can get panic for belopw scenarios.
 // 1. if user does not provide k8s config
 // 2. if controller runtime client instantiation fails.
-func Res(cfg *rest.Config) *Resources {
+func NewResource(cfg *rest.Config) *Resources {
```

Comment:
NewResources because the type is named Resources? Same with package naming, should it be `resources`?  Maybe for another follow-up

Reply:
I think the struct type should be renamed to Resource. Any reason to name type as Resources? In my opinion slices/array type names should be plural, but struct represents a single user-defined type.

Reply:
I wanted to do a fluent style API and suggested the panic.
I am ok if we change it to return an error and let callee decide what to do with error.

Reply:
Renamed `NewResource` to `New` as package name is resources. Also removed panic from `New` instead made it to return an error and let caller decided how to handle the error.